Join existing shopping list by sharing token

a-maffei opened the pull request:
## Description

This code adds a new form with a text input to the Home component. The user can input a token in the form - once the token is submitted, we check our Firebase database to see if there's a list with that same token. If there is a matching token and the list is not empty, the user is redirected to the corresponding shopping list in "/list", and the token is saved to localStorage. If the token doesn't correspond to a shopping list with at least one item, an error message will display telling the user the list does not exist or is empty. If a user has already a token, they are automatically redirected to the List view (which is a feature we already implemented last week).

## Related Issue

closes #5 

## Acceptance criteria
If a user **doesn’t** already have a token:
- [x] The `Home` view shows a form that allows the user to enter a token to join an _existing_ list, in addition to the button that allows them to create a _new_ list.
- [x] The input that accepts the list token has a semantic `label` element associated with its
- [x] The user can submit this form with both the mouse and the `Enter` key
- [x] If the list exists,
	- the token is saved in `localStorage`
	- the user is redirected to the `List` view and shown the items on that list
- [x] If the list _does not_ exist, the user is shown an error message that explains the problem

If a user **does** already have a token:
- [x] They are automatically redirected to the `List` view.

## Type of Changes

<!-- Put an `✓` for the applicable box: -->

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |


## Testing Steps / QA Criteria

- Remove any list token you might have in local storage so you can visit the homepage (path "/")
- In the text box, on the homepage, click on the text "Share token": this should place the focus on the text box so you can start typing
- Type "my test list" and press enter or the button below to submit
- If the code works, you should be redirected to "/list" and see the shopping list connected to the token
- Now remove the token from local storage and type "my list test"
- You should see an error message claiming that the shopping list connected to that token doesn't exist or is empty.

iblas1 committed:
add logic to share existing token

iblas1 committed:
add logic to share existing token new

naura1835 commented on src/App.jsx:
```diff
@@ -58,7 +58,7 @@ export function App() {
 						path="/list"
 						element={<List data={data} listToken={listToken} />}
 					/>
-					<Route path="/add-item" element={<AddItem listId={listToken} />} />
```
Is there a reason why listToken is not being passed as a prop to `AddItem` and `Layout` ?

naura1835 reviewed (changes requested):
Everything looks great

a-maffei committed:
Add missing props to App.js

a-maffei committed:
Remove space between Routes in App.js

a-maffei commented:
@naura1835 I implemented your suggestion. Not sure why the props weren't in App.js anymore, but I added them now.

@iblas1 FYI: I didn't change anything in our code, just passed again two props that had gone "missing" from the App component.

Peacesandy reviewed (approved):
Clean code, I've learnt a particular concept by going through the codebase

a-maffei commented:
Hi @Amaka202, can you please review mine and @iblas1's PR when you have the chance? Thank you!

Amaka202 commented on src/views/Home.jsx:
```diff
@@ -12,12 +17,50 @@ export function Home({ setListToken, listToken }) {
+				/>
+				<button type="submit">Join an existing list</button>
+			</form>
+			{errorMessage && errorMessage}
```
It might be good for accessibility to use a html tag to render the error message eg:
```
{
     errorMessage &&
     <p>{errorMessage}</p>
}
```

Amaka202 commented on src/views/Home.jsx:
```diff
@@ -12,12 +17,50 @@ export function Home({ setListToken, listToken }) {
 		return <Navigate to="/list" />;
 	}
 
+	const submitShareToken = (e) => {
+		e.preventDefault();
+
+		if (tokenInput.length === 0) {
+			setErrorMessage('Please enter a token');
+			setTimeout(() => {
+				setErrorMessage('');
+			}, 3000);
+			return;
+		}
+
+		streamListItems(tokenInput, (snapshot) => {
+			const nextData = getItemData(snapshot);
+
+			if (nextData.length > 0) {
+				setListToken(tokenInput);
+			} else {
+				setErrorMessage('The list you entered is empty or it does not exist');
+				setTimeout(() => {
+					setErrorMessage('');
+				}, 3000);
+			}
+		});
+	};
+
```
This is a beautiful algorithm you wrote here! weldone guys!!💥

Amaka202 reviewed (approved):
Weldone guys, your code looks good!

I left a little suggestion that will improve code quality.✅💚
